### PR TITLE
Error handling of MatrixConstrainedLinearRegression

### LINKED ISSUE #4: AttributeError: 'MatrixConstrainedLinearRegression' object has no attribute '_preprocess_data'
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AttributeError: 'MatrixConstrainedLinearRegression' object has no attribute '_preprocess_data'
```

Reproduce
```python
from constrained_linear_regression import MatrixConstrainedLinearRegression
import numpy as np
from sklearn.datasets import make_regression

# Create a toy regression dataset
X, y = make_regression(n_samples=100, n_features=4)

# Create a constraint matrix and vector
# Here as an example, we give A as an identity matrix and B as ones,
# meaning we want all coefficients to be less than or equal to 1.
A = np.eye(X.shape[1])
B = np.ones(X.shape[1])

# Create and fit the model
model = MatrixConstrainedLinearRegression(A=A, B=B)
model.fit(X, y)
```

### PR DESCRIPTION
**Pull request by ccomkhj**
Fixes #4 : AttributeError: 'MatrixConstrainedLinearRegression' object has no attribute '_preprocess_data' is fixed.

### CODE REVIEW

File: constrained_linear_regression/__init__.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ def fit(self, X, y, initial_beta=None):
             accept_sparse=['csr', 'csc', 'coo'],
             y_numeric=True, multi_output=False
         )
-        X, y, X_offset, y_offset, X_scale = self._preprocess_data(
```
Comment (avidale) [inline constrained_linear_regression/__init__.py:103]: With which version of scikit-learn did you test this?
The method `self._preprocess_data` used to work for me when I implemented this. 
If `scikit-learn` API has changed at some moment, we probably need to find the version where it happened and reflect it in the requirements of `constrained-linear-regression`.

Comment (avidale) [reply constrained_linear_regression/__init__.py:103]: I am currently using `scikit-learn==1.2.2`, and it works fine.

Comment (ccomkhj) [reply constrained_linear_regression/__init__.py:103]: First, when I upgrade sciket-learn>1.1, this error happens.
```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
constrained-linear-regression 0.0.4 requires scikit-learn<=1.1, but you have scikit-learn 1.2.2 which is incompatible.
```
```bash
pip install -U scikit-learn==1.2.2
pip list | grep -E "(constrained|scikit)"
```
```
constrained-linear-regression 0.0.4
scikit-learn                  1.2.2
```
Run the reproducing error code at #4 
```error
X, y, X_offset, y_offset, X_scale = self._preprocess_data(
AttributeError: 'MatrixConstrainedLinearRegression' object has no attribute '_preprocess_data'
```

**Comment by ccomkhj** [comment]
I accidentally include additional commits into this PR, so I'll create another PR.

**Pull request closed without merging**